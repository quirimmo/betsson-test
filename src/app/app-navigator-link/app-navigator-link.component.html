<nav>
	<a class="back-navigation-wrapper hvr-sweep-to-right" routerLink="/">
		<mat-icon class="back-navigation-arrow">arrow_left</mat-icon>
		<span class="back-navigation-text">All Movies</span>
	</a>
</nav>
