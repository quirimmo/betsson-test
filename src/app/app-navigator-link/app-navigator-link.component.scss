@import './../../colors.scss';
.back-navigation-wrapper {
	color: $blue;
	position: relative;
	text-transform: uppercase;
	font-size: 15pt;
	letter-spacing: 0.5px;
	top: -20px;
	left: -50px;
	display: block;
	clear: both;
	height: 110px;
	width: 0px;
	opacity: 0.75;
	-webkit-transition: all 0.35s;
	transition: all 0.35s;
}

.back-navigation-wrapper:hover {
	width: 250px;
	border-bottom: 1px solid white;
	opacity: 1;
	color: white;
}

.back-navigation-arrow {
	font-size: 100pt;
	position: absolute;
	display: inline-block;
	top: 0;
	left: 0;
}

.back-navigation-text {
	position: absolute;
	width: 150px;
	text-align: left;
	top: 50px;
	left: 100px;
}

.hvr-sweep-to-right:before {
	background: $blue;
}
