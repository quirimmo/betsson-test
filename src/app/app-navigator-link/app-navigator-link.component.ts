import { Component } from '@angular/core';
import { MoviesRouterService } from '../movies/movies-router.service';

@Component({
	selector: 'app-navigator-link',
	templateUrl: './app-navigator-link.component.html',
	styleUrls: ['./app-navigator-link.component.scss']
})
export class AppNavigatorLinkComponent {
	constructor(private moviesRouterService: MoviesRouterService) {}

	public navigateBackToMoviesList(): void {
		this.moviesRouterService.navigateBackToMoviesList();
	}
}
