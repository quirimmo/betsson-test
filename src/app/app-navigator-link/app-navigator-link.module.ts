import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { AppNavigatorLinkComponent } from './app-navigator-link.component';
import { MatIconModule } from '@angular/material';

@NgModule({
	imports: [CommonModule, RouterModule, MatIconModule],
	declarations: [AppNavigatorLinkComponent],
	exports: [AppNavigatorLinkComponent],
	providers: []
})
export class AppNavigatorLinkModule {}
