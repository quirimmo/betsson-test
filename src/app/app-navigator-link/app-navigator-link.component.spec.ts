import { TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { MatIconModule } from '@angular/material';
import { AppNavigatorLinkComponent } from './app-navigator-link.component';
import { MoviesRouterService } from '../movies/movies-router.service';

const mockNavigateBackToMoviesList = jest.fn();
const mockMoviesRouterService = {
	provide: MoviesRouterService,
	useValue: {
		navigateBackToMoviesList: mockNavigateBackToMoviesList
	}
};
const providers = [mockMoviesRouterService];
const imports = [RouterTestingModule, MatIconModule];
const declarations = [AppNavigatorLinkComponent];

describe('AppNavigatorLink Component', () => {
	let componentFixture, componentInstance;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations,
			providers,
			imports
		}).compileComponents();
		componentFixture = TestBed.createComponent(AppNavigatorLinkComponent);
		componentInstance = componentFixture.debugElement.componentInstance;
	}));
	afterEach(() => {
		jest.clearAllMocks();
		jest.restoreAllMocks();
	});

	it('should create the component', () => {
		expect(componentFixture).toBeTruthy();
	});

	it('should match the snapshot', () => {
		expect(componentFixture).toMatchSnapshot();
	});

	describe('navigateBackToMoviesList', () => {
		it('should call the navigateBackToMoviesList method of movieRouterService', () => {
			componentInstance.navigateBackToMoviesList();
			expect(mockNavigateBackToMoviesList).toHaveBeenCalled();
		});
	});
});
