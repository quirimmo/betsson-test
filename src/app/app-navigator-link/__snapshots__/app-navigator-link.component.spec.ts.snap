// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigatorLink Component should match the snapshot 1`] = `
<app-navigator-link
  moviesRouterService={[Function Object]}
>
  <nav>
    <div
      class="back-navigation-wrapper hvr-sweep-to-right"
    >
      <mat-icon
        aria-hidden="true"
        class="back-navigation-arrow mat-icon"
        role="img"
      >
        arrow_left
      </mat-icon>
      <span
        class="back-navigation-text"
      >
        All Movies
      </span>
    </div>
  </nav>
</app-navigator-link>
`;
